src/components/results/ResultTables.js

```diff
@@ -0,0 +1,85 @@
+import {
+  CircularProgress,
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow
+} from "@mui/material";
+import {hooks} from "../../config/queryClient";
+import {useContext} from "react";
+import {QuizContext} from "../context/QuizContext";
+import {getAppDataByQuestionId, getDataWithId} from "../context/utilities";
+
+const ResultTables = () => {
+
+  const {data: responses, isLoading} = hooks.useAppData();
+  const {order, questions} = useContext(QuizContext);
+
+
+  if (isLoading) {
+    return <CircularProgress/>;
+  }
+
+
+  /**
+   * Helper function to get the name of all users that have answered to at least one question.
+   *
```

Comment:
I like the comments aha I don't oblige you to do them, but it's indeed really good practice.